Source/WebCore/platform/network/DNS.cpp

```diff
@@ -80,6 +80,19 @@ bool IPAddress::containsOnlyZeros() const
     }), m_address);
 }
 
+bool IPAddress::isLoopback() const
+{
+    return WTF::switchOn(m_address,
+        [] (const struct in_addr& address) {
+        return address.s_addr == INADDR_LOOPBACK;
+    }, [] (const struct in6_addr& address) {
+        return !memcmp(&address.s6_addr, "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\1", sizeof(address.s6_addr));
```

Comment:
Any suggestion for making this more reasonable?

Reply:
Could you use `in6addr_loopback` ?